ui/apps/platform/src/services/GroupsService.ts

```diff
@@ -0,0 +1,53 @@
+import { Group } from 'types/group.proto';
+
+import axios from './instance';
+import { Empty } from './types';
+
+const url = '/v1/groups';
+const updateUrl = '/v1/groupsbatch';
+
+/**
+ * Fetches list of groups of rules
+ */
+export function fetchGroups(): Promise<{ response: { groups: Group[] } }> {
+    return axios.get<{ groups: Group[] }>(url).then((response) => ({
+        response: response.data,
+    }));
+}
+
+/**
+ * Update/Add a group rule.
+ */
+export function updateOrAddGroup({ oldGroups, newGroups }: UpdateGroupsArg): Promise<Empty> {
```

Comment:
nit: the weird thing with this `groupsbatch` API is that it can be used for adding, updating, or removing groups. It would be best to reflect that in the function name, since now it only implies adding / updating.